pokemon-react-app/src/components/InputComponent.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react'
+import { Input } from '../styles/Input.style'
+import { useSelector, useDispatch } from 'react-redux'
+import { filterPokemons } from './../redux/actions/pokemonActions'
+
+function InputComponent() {
+
+    const [searchName, setSearchName] = useState('')
+
+    const pokemons = useSelector(state => state.pokemons.pokemons)
+    const dispatch = useDispatch();
+
+    const searchHandler = (e) => {
+        setSearchName(e.target.value)
+        if (e.target.value !== '') {
+            const filteredPokemons = pokemons.filter((pokemon) => {
+                //return pokemon.name.substring(0,e.target.value.length)===e.target.value.toLowerCase()
+                return pokemon.name.toLowerCase().includes(e.target.value.toLowerCase())
+            })
+            dispatch(filterPokemons(filteredPokemons))
+
+        }
+        else {
+            dispatch(filterPokemons(pokemons))
+        }
+    }
+
+    return (
+        <div>
+            <Input
```

Comment:
You need to implement here a debounce mechanism for the input, because the search performance is very bad. If I started to type something on the input it will freeze the UI after couple of key strokes, because too many actions are dispatched.
Why debounce, check [here](https://css-tricks.com/debouncing-throttling-explained-examples/).